packages/block-editor/src/components/segmented-text-control/index.js

```diff
@@ -41,22 +44,28 @@ export default function SegmentedTextControl( {
 				className
 			) }
 		>
-			<BaseControl.VisualLabel as="legend">
-				{ label }
-			</BaseControl.VisualLabel>
 			<div className="block-editor-segmented-text-control__buttons">
-				{ options.map( ( option ) => {
-					return (
-						<Button
-							size="compact"
-							key={ option.value }
-							icon={ option.icon }
-							label={ option.label }
-							isPressed={ option.value === value }
-							onClick={ () => onChange( option.value ) }
-						/>
-					);
-				} ) }
+				<ToggleGroupControl
+					__nextHasNoMarginBottom
+					__next40pxDefaultSize
+					isDeselectable
+					label={ label }
+					onChange={ ( newValue ) => {
+						onChange( newValue );
+					} }
+					value={ value }
+				>
+					{ options.map( ( option ) => {
+						return (
+							<ToggleGroupControlOptionIcon
+								key={ option.value }
+								value={ option.value }
+								icon={ option.icon }
+								label={ option.label }
+							/>
+						);
+					} ) }
+				</ToggleGroupControl>
 			</div>
 		</fieldset>
```

Comment:
The ToggleGroupControl usage looks good, thank you!

However, the outer `fieldset` and `div` are now completely unnecessary, and this `SegmentedTextControl` component is also unnecessary, as consumers can just use `ToggleGroupControl` directly. There is nothing that this component adds on top of ToggleGroupControl itself.

Ideally, we would clean that up in this PR, but if you prefer, we can split out the refactoring work to a separate PR.  At minimum, I think we should to remove the `fieldset` and `div` before landing.

Reply:
I can update on this PR itself, let me know what suits better. I am open to both.

Thanks

Reply:
Probably better to have those updates as part of this PR, if you don't mind

Reply:
The `SegmentedTextControl` component is a wrapper component introduced in #60841 to make the same code common.

We should be able to simply replace the `SegmentedTextControl` component with the `ToggleGroupControl` component.
Also, since the `SegmentControl` component is not exposed, it is safe to delete it 👍

Reply:
Hi Team, apologies, I would be updating these feedbacks by tomorrow. ✨

Reply:
Hi @mirka @t-hamano @ciampo, Have updated the PR with the requested changes.

Thank You,